data/transactions/logic/eval.go

```diff
@@ -2442,8 +2606,14 @@ func opGaid(cx *EvalContext) {
 
 func opGaids(cx *EvalContext) {
 	last := len(cx.stack) - 1
-	groupIdx := cx.stack[last].Uint
-	sv, err := opGaidImpl(cx, groupIdx, "gaids")
+
+	gi := cx.stack[last].Uint
+
+	if gi >= uint64(len(cx.TxnGroup)) {
+		cx.err = fmt.Errorf("gaids lookup TxnGroup[%d] but it only has %d", gi, len(cx.TxnGroup))
+		return
+	}
+	sv, err := opGaidImpl(cx, int(gi), "gaids")
```

Comment:
Not quite.  This relates to why I changed the opTxnImpl function that takes wide types.  Notice that `opGaidImpl` takes an `int` gi.  So, before we even call it, we must at least check that the uint64 gi we have here fits in a u32, or else it might wrap around when cast.  It might even wrap around to a nice small positive number, and not cause an error in the impl.  So we check it first.

This complexity is what I was trying to fix by the way I redid opTxnImpl, so it can accept the wider types, and do all the checking there.  I could change gaid opcodes to work in a similar way.

Reply:
I've change opGaidImpl to work the same way as opTxnImpl.  It takes uint64 and all the checking is now there in one place.